src/hooks/useAuth.ts

```diff
@@ -0,0 +1,53 @@
+import { useCallback } from 'react'
+import { useWeb3React, UnsupportedChainIdError } from '@web3-react/core'
+import { NoBscProviderError } from '@binance-chain/bsc-connector'
+import {
+  NoEthereumProviderError,
+  UserRejectedRequestError as UserRejectedRequestErrorInjected,
+} from '@web3-react/injected-connector'
+import {
+  UserRejectedRequestError as UserRejectedRequestErrorWalletConnect,
+  WalletConnectConnector,
+} from '@web3-react/walletconnect-connector'
+import { connectorLocalStorageKey, ConnectorNames } from '@pancakeswap-libs/uikit'
+import { useToast } from 'state/hooks'
+import { connectorsByName } from 'connectors'
+
+const useAuth = () => {
+  const { activate, deactivate } = useWeb3React()
+  const { toastError } = useToast()
+
+  const login = useCallback((connectorID: ConnectorNames) => {
+    const connector = connectorsByName[connectorID]
+    if (connector) {
+      activate(connector, async (error: Error) => {
+        window.localStorage.removeItem(connectorLocalStorageKey)
+        if (error instanceof UnsupportedChainIdError) {
+          toastError('Unsupported Chain Id', 'Unsupported Chain Id Error. Check your chain Id.')
+        } else if (error instanceof NoEthereumProviderError || error instanceof NoBscProviderError) {
+          // eslint-disable-next-line no-console
+          console.log("mememem")
```

Comment:
????

Reply:
sorry, it was a leftover for debugging . I will remove it